src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -549,6 +553,17 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
                     Err(()) => return
                 }
             }
+            name if name.starts_with("atomic_element_") => {
+                let ty = substs.type_at(0);
+                if int_type_width_signed(ty, self).is_some() {
```

Comment:
This restricts the type to integer types. I'm not sure if this is the best action, or if we should accept any type with a size that is a power of two less than the target-specific atomic size limit. But this is what we do for other atomic operations, so it is my first inclination.